### PR TITLE
Chapter 4 5

### PR DESCRIPTION
**Pull request by Moonforeva**
Ready

### CODE REVIEW

**Comment by Moonforeva** [comment]
Ready

File: Gemfile.lock [hunk 1]
```diff
@@ -59,9 +59,17 @@ GEM
     addressable (2.7.0)
       public_suffix (>= 2.0.2, < 5.0)
     ast (2.4.1)
+<<<<<<< HEAD
```
Comment (ngocvt-0484) [inline Gemfile.lock:62]: please fix conflict here

File: app/helpers/users_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (ngocvt-0484) [inline app/helpers/users_helper.rb:1]: remove this file because you have not use

File: app/views/layouts/_footer.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <small>
+    The <a href="https://www.railstutorial.org/">Ruby on Rails Tutorial</a>
```
Comment (ngocvt-0484) [inline app/views/layouts/_footer.html.erb:3]: use link_to

**Comment by Moonforeva** [comment]
Ready

File: app/controllers/static_pages_controller.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 class StaticPagesController < ApplicationController
   def home; end
-  
   def help; end
+  def about; end
+  def contact; end
```
Comment (chungpv-1008) [inline app/controllers/static_pages_controller.rb:5]:
```suggestion

  def about; end

  def contact; end
```

File: app/views/layouts/_shim.html.erb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<!--[if lt IE 9]>
```
Comment (chungpv-1008) [inline app/views/layouts/_shim.html.erb:1]: remove all comments and add end line for this file

File: app/views/static_pages/about.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+<p>
+  The <%= link_to "Ruby on Rails Tutorial", Settings.links.rails_tutorial.home_page %>, part of the
+  <%= link_to "Learn Enough", Settings.links.other_website.learn_enough %> family of
+  tutorials, is a
+  <%= link_to "book", Settings.links.rails_tutorial.book_page %>
+  <%= link_to "screencast series", Settings.links.rails_tutorial.screencast_series %>
+  to teach web development with
+  <%= link_to "Ruby on Rails", Settings.links.rails_tutorial.home_page %>.
+  This is the sample app for the tutorial.
+</p>
```
Comment (chungpv-1008) [inline app/views/static_pages/about.html.erb:13]: add end line for this file

File: app/views/static_pages/contact.html.erb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<%= t provide :title, "Contact" %>
```
Comment (chungpv-1008) [inline app/views/static_pages/contact.html.erb:1]:
```suggestion
<%= provide :title, t("Contact") %>
```

Comment (chungpv-1008) [inline app/views/static_pages/contact.html.erb:1]:
```suggestion
<%= provide :title, "Contact" %>
```

File: app/views/users/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<%= t provide :title, "Sign Up" %>
```
Comment (chungpv-1008) [inline app/views/users/new.html.erb:1]:
```suggestion
<%= provide :title, "Sign Up" %>
```

File: config/routes.rb [hunk 9]
```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
   root "static_pages#home"
-  get "static_pages/help"
+  get   "/home",    to: "static_pages#home"       
+  get   "/help",    to: "static_pages#help"
+  get   "/about",   to: "static_pages#about"
+  get   "/contact", to: "static_pages#contact"
+  get   "/signup",  to: "user#new"
```
Comment (chungpv-1008) [inline config/routes.rb:7]:
```suggestion

  get   "/home", to: "static_pages#home"       
  get   "/help", to: "static_pages#help"
  get   "/about", to: "static_pages#about"
  get   "/contact", to: "static_pages#contact"
  get   "/signup", to: "user#new"
```

**Comment by Moonforeva** [comment]
Ready

**Review by chungpv-1008 — COMMENTED** [review]
merged

**Pull request merged**